src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -57,5 +83,142 @@ public void SendAsync_SimpleGet_Success()
+            );
+            handler.Dispose();
+        }
+
+        [Fact]
+        public void UriRedirectionWithNetworkCredential()
+        {
+            var redirectUri = HttpTestServers.RedirectUriForCreds(Username, Password);
+            var handlerWithNC = new HttpClientHandler();
+            handlerWithNC.Credentials = _credential;
+            var unAuthClient = new HttpClient(handlerWithNC);
+            var unAuthResponse = unAuthClient.GetAsync(redirectUri).Result;
+            Assert.Equal(HttpStatusCode.Unauthorized, unAuthResponse.StatusCode);
+            handlerWithNC.Dispose();
+            unAuthClient.Dispose();
+       }
+
+        [Fact]
+        public void UriRedirectionCredentialCache()
+        {
+            var uri = HttpTestServers.BasicAuthUriForCreds(Username, Password);
+            var redirectUri = HttpTestServers.RedirectUriForCreds(Username, Password);
+            var credentialCache = new CredentialCache();
+            credentialCache.Add(uri, "Basic", _credential);
+            var handler = new HttpClientHandler();
+            handler.Credentials = credentialCache;
+            var client = new HttpClient(handler);
+            var response = client.GetAsync(uri).Result;
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            handler.Dispose();
+            client.Dispose();
+        }
+
+        [Theory]
+        [InlineData("cookiename", "cookievalue")]
+        public void CookieTest(string name, string value)
+        {
+            var handler = new HttpClientHandler();
+            Assert.True(handler.UseCookies);
+            var cookieContainer = new CookieContainer();
+            cookieContainer.Add(HttpTestServers.RemoteServerCookieUri, new Cookie(name, value));
+            handler.CookieContainer = cookieContainer;
+            var client = new HttpClient(handler);
+            var httpResponse = client.GetAsync(HttpTestServers.RemoteServerCookieUri).Result;
+            Assert.Equal(httpResponse.StatusCode, HttpStatusCode.OK);
+            var response = httpResponse.Content.ReadAsStringAsync().Result;
+
+            Assert.True(JsonMessageContainsKeyValue(response, name, value));
+            handler.Dispose();
+            client.Dispose();
+        }
+
+        [Theory]
+        [InlineData("X-Cust-Header","x-value")]
+        public void HeadersTest(string name, string value)
+        {
+            var handler = new HttpClientHandler();
+            var client = new HttpClient(handler);
+            client.DefaultRequestHeaders.Add(name, value);
+            var httpResponse = client.GetAsync(HttpTestServers.RemoteServerHeadersUri).Result;
+            Assert.Equal(httpResponse.StatusCode, HttpStatusCode.OK);
+            var response = httpResponse.Content.ReadAsStringAsync().Result;
+            Assert.True(JsonMessageContainsKeyValue(response, name, value));
+            handler.Dispose();
+            client.Dispose();
+        }
```

Comment:
Since the primary initial consumer of this component will be NuGet package restore, is there a test we can add that uses HttpClient in the same way it does, similar combinations of features, etc.?  I think it's this: https://github.com/aspnet/dnx/blob/22dbb542f72158f20d0af7eba8abd4bea5a14d22/src/Microsoft.Dnx.Tooling/Restore/NuGet/HttpSource.cs